client/src/app/components/task-modal/task-modal/task-modal.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, inject } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { NgbActiveModal, NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ITask } from 'src/app/interfaces/task';
+
+@Component({
+  selector: 'app-task-modal',
+  templateUrl: './task-modal.component.html',
+  styleUrls: ['./task-modal.component.scss'],
+})
+export class TaskModalComponent {
+  @Input() task!: ITask;
```

Comment:
I'm a little confused by `task!: ITask`: should it be `task: ITask | null`? I'm assuming that a missing task means it's a create modal, but if there's a task it's for editing.

We definitely want TypeScript to know that `task` can be null/undefined. I'm not super familiar with the `task!: ITask` syntax: does TypeScript know that `task` can be null/undefined, and you're just saying that the type must be `ITask` if the input is present?